googletest/include/gtest/internal/gtest-port.h

```diff
@@ -682,12 +682,12 @@ typedef struct _RTL_CRITICAL_SECTION GTEST_CRITICAL_SECTION;
 // A macro to disallow move operator=
 // This should be used in the private: declarations for a class.
 #define GTEST_DISALLOW_MOVE_ASSIGN_(type) \
-  type& operator=(type &&) = delete
+  type& operator=(type &&) noexcept = delete
 
```

Comment:
Hi @asoffer, @N-Dekker.
Main motivation is to fix one sonarqube smell/bug in this area. At the moment I don't have an access to verify which exactly rule wad triggered, but after searching in SonarQube rules that one https://rules.sonarsource.com/cpp/RSPEC-5018?search=Noexcept looks looks the most familiar. 
It's supported by https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#c66-make-move-operations-noexcept

Indeed it looks odd to have deleted function noexcept, but still it's a part of API.

I'm not strongly pushing to have this in. Just wanted to clean all smells from generated code.

Reply:
@trzecieu I'm not _opposed_ to declaring deleted move-member-functions `noexcept`, it's just that I had not seen it before! While it's a common practice to declare a deleted assignment operator to return `type&`, instead of `void`, even though it will never return anything at all!

So you could argue that adding `noexcept` improves consistency between deleted and non-deleted move-member-functions. I leave it up you. Either way is acceptable to me, at the moment.

Reply:
I'm accepting the PR as is, because this is not a particularly important style point. But I've summarized my thoughts below, because I do think the broader issue of style-guide conformance and consistency is an important discussion to have.

The google style guide also suggests marking move-constructors as noexcept. What I think is different here is that all of these guidelines are about move constructors that are actually present in code. The reasoning is typically "moves are about performance" which doesn't apply for deleted move ctors/assignments.

The return value should be `T&` for move-assignment, because that's part of how people syntactically recognize that it is a move-assignment operator. The noexcept specification is not part of that recognition, which makes me think it's less important. I don't think consistency here is important, but rather being able to quickly identify the two important pieces: "move operator" and "deleted."

Anyway, as I said, I'm accepting the PR as is, so please don't make any further commits. Otherwise our automated machinery will get confused.